netsim-ffi/src/lib.rs

```diff
@@ -169,17 +171,21 @@ pub unsafe extern "C" fn netsim_socket_recv(
     let Some((id, data)) = socket.recv() else {
         // this is usually to signal it is time to release
         // the socket
-        todo!()
+        let _ = Box::from_raw(socket);
+        return SimError::Undefined
```

Comment:
Arf, if you do that you kind of break the expected behaviour I think. The expected behaviour would be to say something like "SimError::NoMoreData" or something like that. And to let the user call `sim_socket_delete_release` when hey are ready.